packages/marketprice/src/alarms/mod.rs

```diff
@@ -262,29 +258,12 @@ where
     }
 
     pub fn last_delivered(&mut self) -> Result<(), AlarmError> {
-        self.in_delivery
-            .pop_front(self.storage.deref_mut())
-            .map_err(AlarmError::LastDeliveredRemove)
-            .and_then(|maybe_alarm: Option<AlarmWithSubscriber>| {
-                maybe_alarm.map(|_: AlarmWithSubscriber| ()).ok_or_else(|| {
-                    AlarmError::EmptyAlarmsInDeliveryQueue(String::from(
-                        "Received success reply status",
-                    ))
-                })
-            })
+        self.pop_front_in_delivery("Received success reply status")
+            .map(|_: AlarmWithSubscriber| ())
     }
 
     pub fn last_failed(&mut self) -> Result<(), AlarmError> {
-        self.in_delivery
-            .pop_front(self.storage.deref_mut())
-            .map_err(AlarmError::LastFailedRemove)
```

Comment:
I am afraid we misled the caller by sending back Error::LastDeliveredRemove instead

Reply:
Good catch!